Change styles related to loop-flow

flomnes opened the pull request:
Fix #565 

* Change red (error) to orange (warning)
* Warning only if loop flow is enabled
* Cell style "warning" for transmission capacites if rule is violated

flomnes committed:
Change styles related to loop-flow

* Change red (error) to orange (warning)
* Warning only if loop flow is enabled
* Cell style "warning" for transmission capacites if rule is violated

guilpier-code commented on src/ui/simulator/toolbox/components/datagrid/renderer/connection.h:
```diff
@@ -92,6 +92,7 @@ class Connection : public Renderer::Matrix<>
 
 protected:
     wxWindow* pControl;
+    bool useLoopFlow = false;
```
As **useLoopFlow** is a member variable, its name should reflect it (**useLoopFlow_** or **pUseLoopFlow**)

flomnes replied:
done

guilpier-code commented on src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp:
```diff
@@ -137,26 +139,44 @@ bool connectionParameters::cellValue(int x, int y, const Yuni::String& value)
     return true;
 }
 
+static bool checkLoopFlowNTCDirect(double ntcDirect, double loopFlow, bool useLoopFlow)
```
Function **checkLoopFlowNTCDirect** could be called inside function **checkLoopFlow**.
Same thing for **checkLoopFlowNTCIndirect**.

flomnes replied:
done

guilpier-code commented on src/ui/simulator/toolbox/components/datagrid/renderer/connection.h:
```diff
@@ -137,9 +138,11 @@ class connectionNTCdirect : public connectionNTC
 
 private:
     void setMatrix(Data::AreaLink* link) override;
+    Antares::Matrix<>* parameters;
```
You should make **parameters** a private variable in parent class **connectionNTC**.
You should change **parameters** into **parameters_** or **pParameters**

flomnes replied:
You mean protected ? If private, child classes `connectionNTCindirect` and `connectionNTCdirect` can't access it, unless we be`friend` them.

guilpier-code replied:
Yes, I mean **protected**, sorry. I meant private in the sense protected :-D

guilpier-code commented on src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp:
```diff
@@ -265,6 +298,20 @@ connectionNTCindirect::connectionNTCindirect(wxWindow* parent,
+        return Renderer::Matrix<>::cellStyle(col, row);
+
+    const double loopFlow = (*parameters)[Data::fhlLoopFlow][row];
+    const double ntcInirect = Renderer::Matrix<>::cellNumericValue(col, row);
```
**ntcInirect** should be changed into **ntcIndirect** (a **d** is missing)

flomnes replied:
done

flomnes committed:
PR remarks

flomnes committed:
PR remarks

flomnes committed:
Turn static checkLoopFlow functions into virtual methods

flomnes commented:
I applied all suggested changes. Notice also the small re-factor : I moved most of `cellStyle` one level up into `ConnectionNTC` and moved loopflow check functions (previously static) into `connectionNTCdirect` and `connectionNTCindirect`

flomnes commented on src/ui/simulator/toolbox/components/datagrid/renderer/connection.cpp:
```diff
@@ -241,6 +247,19 @@ bool connectionNTC::cellValue(int x, int y, const Yuni::String& value)
+
+    const double loopFlow = (*mLoopFlowData)[row];
+    const double ntc = Renderer::Matrix<>::cellNumericValue(col, row);
+    if (!checkLoopFlow(ntc, loopFlow))
```
pure virtual function, overriden in `connectionNTCdirect` and `connectionNTCindirect`
